lib/shopify_api/auth.rb

```diff
@@ -19,7 +19,7 @@ def embedded_app_url(host)
         end
 
         decoded_host = Base64.decode64(host)
-        "https://#{decoded_host}/apps/#{Context.api_key}"
+        "#{Context.host_scheme}://#{decoded_host}/apps/#{Context.api_key}"
```

Comment:
I think this needs to stay as https - this is us redirecting back to Shopify at this point, which would always be https.